integration_test.go

```diff
@@ -134,11 +134,14 @@ func TestIntegration(t *testing.T) {
 	}
 	s3Service := s3.NewFromConfig(cfg)
 
-	_, err = s3Service.CreateBucket(context.Background(), &s3.CreateBucketInput{
-		Bucket: aws.String("bucket"),
-	})
+	_, err = s3Service.HeadBucket(context.TODO(), &s3.HeadBucketInput{Bucket: aws.String("bucket")})
```

Comment:
Alternatively, could we delete the bucket at the end of each test using a "cleanup" `func()`?

Reply:
That seems like a better idea. Should this HeadBucket check move into `main`?

Reply:
Minio actually forgets about all buckets when it goes away, so the existing cleanup in `TestMain` _should_ be doing the job. But I've found that occasionally minio doesn't properly get killed, so the next run just picks up the existing minio, which is when we get this error.

Rather than checking if the bucket exists, I'd prefer to (a) do a more robust job killing minio, even on panics in the test harness, and (b) on startup check if minio is running and kill it again if necessary.

Reply:
@jsha I just pushed a commit that should address your comment.